timer/timer.go

```diff
@@ -0,0 +1,144 @@
+package timer
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Timer definition
+type (
+	Timer struct {
+		ID             uuid.UUID
+		Domain         string
+		ExecutionCount uint32
+		Meta           Meta
+		Task           Task
+		Schedule       Schedule
+	}
+
+	// Task represents a timer task.
+	Task interface {
+		isTask()
+	}
+
+	// Schedule represents the "when" configuration of a timer.
+	Schedule interface {
+		isSchedule()
+	}
+
+	// Meta represents timer metadata.
+	Meta struct {
+		CreationTime time.Time
+	}
+)
+
+// HTTP Task represents an HTTP request.
+type (
+	HTTP struct {
+		URL string
+		Method
```

Comment:
Why is method embeded but others are not?

Reply:
Ok i'll make it not embedded